File: packages/sdkwork-utils-python/sdkwork_utils/optional.py
from sdkwork_utils.string import is_blank


def coalesce(*values: str | None) -> str | None:
    for value in values:
        if not is_blank(value):
            return value.strip() if value is not None else None
    return None


def default_if_blank(value: str | None, default: str) -> str:
    if is_blank(value):
        return default
    return value.strip() if value is not None else default

File: packages/sdkwork-utils-rust/src/optional.rs
use crate::string::is_blank;

pub fn coalesce<'a>(values: &[Option<&'a str>]) -> Option<&'a str> {
    for value in values {
        if let Some(text) = value {
            if !is_blank(Some(text)) {
                return Some(text.trim());
            }
        }
    }
    None
}

pub fn default_if_blank(value: Option<&str>, default: &str) -> String {
    match value {
        Some(text) if !is_blank(Some(text)) => text.trim().to_string(),
        _ => default.to_string(),
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn optional_helpers() {
        assert_eq!(
            coalesce(&[None, Some(""), Some("  "), Some("ok")]),
            Some("ok")
        );
        assert_eq!(default_if_blank(Some("  "), "fallback"), "fallback");
    }
}
